### ISSUE TITLE
[Serious Bug] Wrong records being retrieved from the OneToMany relationship

### ISSUE THREAD

**Issue by jasonlaw** [rivantsov/vita#240]
I have the following entities:

~~~C#
[Entity]
public interface ICmpany
{
   // New Rewards Campaign
   [OneToMany(thisEntityRef: "Company")] IList<IRewardsCampaign> RewardsCampaigns { get; }
   [OneToMany(thisEntityRef: "AsActive")] IList<IRewardsCampaign> ActiveRewardsCampaigns { get; }
   [OneToMany(thisEntityRef: "AsReport")] IList<IRewardsCampaign> ReportRewardsCampaigns { get; }
   [OneToMany(thisEntityRef: "AsPublished")] IList<IRewardsCampaign> PublishedRewardsCampaigns { get; }
   [OneToMany(thisEntityRef: "AsScheduled")] IList<IRewardsCampaign> ScheduledRewardsCampaigns { get; }
}

[Entity]
public interface IRewardsCampaign
{
    [NoUpdate] ICompany Company { get; set; }
    [Nullable] ICompany AsReport { get; set; }
    [Nullable] ICompany AsActive { get; set; }
    [Nullable] ICompany AsPublished { get; set; }
    [Nullable] ICompany AsScheduled { get; set; }
}
~~~

I set the Company reference to the AsXXX field in IRewardsCampaign entity according to the status.
When testing, with multiple campaign records being added with different status for a Company, I found that all the relations return me the same results, I think this is a serious issue as it leads to logic flaw.

Thanks.

**Comment by rivantsov** [comment]
Should be fixed now, v4.3, packages pushed. Sources will be updated soon, having trouble with git remotes

**Comment by jasonlaw** [comment]
Tested and working now, thank you very much!
